### PR TITLE
Untangle precommit and travis test scripts

### PR DESCRIPTION
**Pull request by erikjohnston**
The main motivation for this is that we want to be able to configure the linters depending on if they're run by travis (use all the CPU and RAM) or locally (avoid locking up the computer)

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Dev Scripts
+
+These are a collection of scripts that should be helpful for those developing
+on dendrite.
+
+- `./scripts/find-lint.sh` runs the linters against dendrite,
```
Comment (richvdh) [inline scripts/README.md]: this looks like a list that is going to get out of date in about 30 seconds.

I'd suggest sticking these descriptions in the comments at the top of each script. And possibly duplicating the env var descriptions to each one?

File: scripts/find-lint.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+go install github.com/alecthomas/gometalinter/
+gometalinter --config=linter.json ./... --install
+
+echo "Looking for lint..."
+gometalinter ./... $args
+
+echo "Double checking spelling..."
+misspell -error src *.md
+
+echo "Done!"
```
Comment (richvdh) [inline scripts/find-lint.sh]: this is going to look silly when it's called from another script

File: scripts/travis-test.sh [hunk 3]
```diff
@@ -13,7 +16,7 @@ gb build github.com/matrix-org/dendrite/cmd/mediaapi-integration-tests
 gb build github.com/matrix-org/dendrite/cmd/client-api-proxy
 
 # Run the pre commit hooks
```
Comment (richvdh) [inline scripts/travis-test.sh]: no longer the pre-commit hooks

File: scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  `./scripts/find-lint.sh fast` runs a subset of faster lints
+- `./scripts/build-test-lint.sh` builds, tests and lints dendrite, and
+  should be run before pushing commits
+- `./scripts/install-local-kafka.sh` downloads, installs and runs a
+  kafka instance
+- `./scripts/travis-test.sh` is what travis runs
+
+
+The linters can take a lot of resources and are slow, so they can be
+configured using two enviroment variables:
```
Comment (richvdh) [inline scripts/README.md]: enviroment

File: scripts/travis-test.sh [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 set -eu
 
+# Tune the GC to use more memory to reduce the number of garbage collections
+export GOGC=400
```
Comment (richvdh) [inline scripts/travis-test.sh]: do we need this as well as DENDRITE_LINT_DISABLE_GC ?

File: .travis.yml [hunk 6]
```diff
@@ -24,8 +24,8 @@ before_script:
  - openssl req -x509 -newkey rsa:4096 -keyout server.key -out server.crt -days 365 -nodes -subj /CN=localhost
 
 script:
- - ./travis-install-kafka.sh
- - ./travis-test.sh
+ - ./scripts/install-local-kafka.sh
```
Comment (richvdh) [inline .travis.yml]: this is linked to as travis-install-kafka from the INSTALL instructions iirc, pls could you have a grep and fix

File: scripts/travis-test.sh (hunk 5 above)
Comment (erikjohnston) [reply scripts/travis-test.sh]: `DENDRITE_LINT_DISABLE_GC` won't disable GC for the unit/integration tests or anything, which I guess is why its there?

**Pull request merged**